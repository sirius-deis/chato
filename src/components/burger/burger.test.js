import { render, screen, fireEvent } from '@testing-library/react';
import Burger from './burger';

describe('Burger component', () => {
  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<Burger onClick={fn}></Burger>);
    expect(container).toMatchSnapshot();
  });
  it('should should fire a click event', () => {
    const fn = jest.fn();
    render(<Burger onClick={fn}></Burger>);
    fireEvent.click(screen.getByRole('button'));
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
