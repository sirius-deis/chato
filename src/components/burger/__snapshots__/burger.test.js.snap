// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger component should match snapshot 1`] = `
<div>
  <button
    class="sc-aXZVg gUJark"
  >
    <div
      class="sc-gEvEer hYBPkA"
    />
    <div
      class="sc-gEvEer hYBPkA"
    />
    <div
      class="sc-gEvEer hYBPkA"
    />
  </button>
</div>
`;
